lib/t/matches.ex

```diff
@@ -894,4 +895,65 @@ defmodule T.Matches do
     )
     |> where([m, e, c], is_nil(c.timestamp))
   end
+
+  def check_matches_calls() do
+    matches_without_successful_calls =
+      Match
+      |> join(:inner, [m], c in Call,
+        on:
+          (c.caller_id == m.user_id_1 and c.called_id == m.user_id_2) or
+            (c.caller_id == m.user_id_2 and c.called_id == m.user_id_1)
+      )
+      |> where([m, c], c.accepted_at >= fragment("now() - interval '80 hours'"))
+      |> select([m], m.id)
+
+    matches_for_push =
+      Timeslot
+      |> where([t], t.selected_slot >= fragment("now() - interval '80 hours'"))
+      |> where([t], t.match_id not in subquery(matches_without_successful_calls))
+      |> select([t], t.match_id)
+
+    good_users =
+      Call
+      |> join(:inner, [c], m in Match,
+        on:
+          (c.caller_id == m.user_id_1 and c.called_id == m.user_id_2) or
+            (c.caller_id == m.user_id_2 and c.called_id == m.user_id_1)
+      )
+      |> where([c, m], c.inserted_at >= fragment("now() - interval '80 hours'"))
+      |> where([c, m], m.id in subquery(matches_for_push))
+      |> where([c, m], c.caller_id == m.user_id_1 or c.caller_id == m.user_id_2)
+      |> select([c, m], c.caller_id)
+      |> T.Repo.all()
+
+    not_in_good_users =
+      Call
+      |> join(:inner, [c], m in Match,
+        on:
+          (c.caller_id == m.user_id_1 and c.called_id == m.user_id_2) or
+            (c.caller_id == m.user_id_2 and c.called_id == m.user_id_1)
+      )
+      |> where([c, m], c.inserted_at >= fragment("now() - interval '80 hours'"))
+      |> where([c, m], m.id in subquery(matches_for_push))
+      |> where([c, m], c.caller_id == m.user_id_1 or c.caller_id == m.user_id_2)
+      |> select([c, m], c.caller_id)
+
+    bad_users =
+      Call
+      |> join(:inner, [c], m in Match,
+        on:
+          (c.caller_id == m.user_id_1 and c.called_id == m.user_id_2) or
+            (c.caller_id == m.user_id_2 and c.called_id == m.user_id_1)
+      )
+      |> where([c, m], c.inserted_at >= fragment("now() - interval '80 hours'"))
+      |> where([c, m], m.id in subquery(matches_for_push))
+      |> where([c, m], c.called_id == m.user_id_1 or c.called_id == m.user_id_2)
+      |> where([c, m], c.called_id not in subquery(not_in_good_users))
```

Comment:
if `not_in_good_users` only used here it can be substituted with `not in ^good_users`